sound/soc/sof/ipc4-topology.c

```diff
@@ -36,6 +36,8 @@ static const struct sof_topology_token ipc4_sched_tokens[] = {
 static const struct sof_topology_token pipeline_tokens[] = {
 	{SOF_TKN_SCHED_DYNAMIC_PIPELINE, SND_SOC_TPLG_TUPLE_TYPE_BOOL, get_token_u16,
 		offsetof(struct snd_sof_widget, dynamic_pipeline_widget)},
+	{SOF_TKN_SCHED_PIPELINE_KCPS, SND_SOC_TPLG_TUPLE_TYPE_WORD, get_token_u32,
+		offsetof(struct snd_sof_widget, pipeline_kcps)},
```

Comment:
it's the first time I hear about a pipeline-level cycle count. We already have a module 'cpc' (cycle per chunk) and 'cps" *cycles per second", so it's not clear to me how those 3 tokens would be used.

Reply:
why we need kcps ?  In current 0004 branch, kcps is built from cpc and we already have cpc in pipeline now. Why do we need to add kcps ?
````c
int kcps = cd->cpc * 1000 / ppl_data->p->period;
````

Reply:
In 0004 branch, kcps is built from each module in pipeline with the above algorithm, why we need pipeline kcps ?
````c
                data.start = p->source_comp;
                data.p = p;
                walk_ctx.comp_func = add_pipeline_cps_consumption;

                if(!clocks_handled)
                        ret = walk_ctx.comp_func(p->source_comp, NULL, &walk_ctx, PPL_DIR_DOWNSTREAM);
````